File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/observer/observer/ObserverOne.java
package com.bigboxcode.designpattern.observer.observer;

public class ObserverOne extends Observer{

    public ObserverOne(Subject subject){
        this.subject = subject;
        this.subject.attach(this);
    }

    @Override
    public void sendUpdate() {
        System.out.println( "Received in ObserverOne: " + subject.getState());
    }
}

File: design-patterns/ts/observer/general/observer-one.ts
// observer-one.ts

import Observer from "./observer";
import Subject from "./subject";


class ObserverOne extends Observer {
    constructor(subject: Subject) {
        super();

        this.subject = subject;
        this.subject.attach(this);
    }

    sendUpdate(): void {
        console.log("Received in ObserverOne: " + this.subject?.getState());
    }
}

export default ObserverOne;